Add a way to create resources in webkitscmpy.GitHub

alancoon opened the pull request:
#### 1c7ed2344ae2d2adf9a781b853e726b5375bb52b
<pre>
Add a way to create resources in webkitscmpy.GitHub
<a href="https://bugs.webkit.org/show_bug.cgi?id=241618">https://bugs.webkit.org/show_bug.cgi?id=241618</a>
rdar://95153858

Reviewed by NOBODY (OOPS!).

Add a new create() method in webkitscmpy&apos;s GitHub SCM class for POST
manipulation of the GitHub API.
Add a new create_reference() method to create git references via API.
Add a new create_tag() method for creating tags in git.

* Tools/Scripts/libraries/webkitscmpy/setup.py: bump webkitscmpy version.
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/__init__.py: bump webkitscmpy version.
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/remote/git_hub.py: add create() and create_tag().
(GitHub.PRGenerator.create): refactor to use the repository&apos;s create() method.
(GitHub.find):
(GitHub):
(GitHub.commits):
(GitHub.create): use for POST actions.
(GitHub.create_reference): leverages create to make an accessible git reference.
(GitHub.create_tag): leverages create and create_reference to make a git tag.
</pre>

alancoon commented on Tools/Scripts/libraries/webkitscmpy/webkitscmpy/remote/git_hub.py:
```diff
@@ -649,3 +641,27 @@ def find(self, argument, include_log=True, include_identifier=True):
             raise ValueError("'{}' is not an argument recognized by git".format(argument))
         return self.commit(hash=commit_data['sha'], include_log=include_log, include_identifier=include_identifier)
+
+    def create(self, path: str, data: dict, headers: dict = None) -> Optional[requests.Response]:
```
Remove `Optional` from typing hint. This will always return `requests.Response`.

JonWBedard replied:
We are still supporting Python 2 in some of the consumers of this library. We need to discuss deprecating Python 2 for those use cases, but that's why our checks still run Python 2 unit tests.

alancoon replied:
Ah I see. In that case are we better off removing typing hints?

JonWBedard replied:
Yes, that's basically why we don't use type hints in these libraries at the moment

alancoon commented on Tools/Scripts/libraries/webkitscmpy/webkitscmpy/remote/git_hub.py:
```diff
@@ -649,3 +641,27 @@ def find(self, argument, include_log=True, include_identifier=True):
+            json=data
+        )
+
+    def create_tag(self, tag, object, message, object_type: str = 'commit') -> Optional[requests.Response]:
```
Ditto about `Optional`

alancoon committed:
Add a way to create resources in webkitscmpy.GitHub
https://bugs.webkit.org/show_bug.cgi?id=241618
rdar://95153858

Reviewed by NOBODY (OOPS!).

Add a new create() method in webkitscmpy's GitHub SCM class for POST
manipulation of the GitHub API.
Add a new create_reference() method to create git references via API.
Add a new create_tag() method for creating tags in git.

* Tools/Scripts/libraries/webkitscmpy/setup.py: bump webkitscmpy version.
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/__init__.py: bump webkitscmpy version.
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/remote/git_hub.py: add create() and create_tag().
(GitHub.PRGenerator.create): refactor to use the repository's create() method.
(GitHub.find):
(GitHub):
(GitHub.commits):
(GitHub.create): use for POST actions.
(GitHub.create_reference): leverages create to make an accessible git reference.
(GitHub.create_tag): leverages create and create_reference to make a git tag.
